samples/Application.Demo/App_Start/MyDemoComposer.cs

```diff
@@ -1,37 +1,26 @@
-﻿using Our.Umbraco.EnvironmentDashboard.Models;
-using Umbraco.Core;
+﻿using System.Collections.Generic;
+using Our.Umbraco.EnvironmentDashboard;
+using Our.Umbraco.EnvironmentDashboard.Components;
+using Our.Umbraco.EnvironmentDashboard.Groups;
 using Umbraco.Core.Composing;
 
-namespace Application.Demo.Start_Up
+namespace Application.Demo
 {
 	public class MyDemoComposer : IUserComposer
 	{
 		public void Compose(Composition composition)
 		{
-			composition.Components().Append<MyDemoComponent>();
-		}
-	}
-
-	public class MyDemoComponent : IComponent
-	{
-		public void Initialize()
-		{
-			// Register as many environments as you wish
-			EnvironmentInfo.Instance.Domains.Add(new DomainInfo("Local", "environmentdashboard.localhost"));
-			EnvironmentInfo.Instance.Domains.Add(new DomainInfo("Development", "environmentdashboard-dev.ry.com"));
-			EnvironmentInfo.Instance.Domains.Add(new DomainInfo("QA", "environmentdashboard-qa.ry.com"));
-			EnvironmentInfo.Instance.Domains.Add(new DomainInfo("UAT", "environmentdashboard-uat.ry.com"));
-			EnvironmentInfo.Instance.Domains.Add(new DomainInfo("Production", "environmentdashboard.ry.com"));
+			composition.AddEnvironmentDashboard(new Dictionary<DashboardEnvironment, IEnumerable<string>>
+			            {
+				            {new DashboardEnvironment("Local"), new[] {"localhost:21801"}},
+				            {new DashboardEnvironment("Development"), new[] {"environmentdashboard-dev.ry.com"}},
+				            {new DashboardEnvironment("QA"), new[] {"environmentdashboard-qa.ry.com"}},
+				            {new DashboardEnvironment("UAT"), new[] {"environmentdashboard-uat.ry.com"}},
+				            {new DashboardEnvironment("Production"), new[] {"environmentdashboard.ry.com"}},
+			            })
+			           .Append<UptimeDashboardGroupsProvider>()
+			           .Append<DatabaseServerGroupsProvider>();
```

Comment:
I like the ideas here, and think that we can streamline it more to make it easier for the implementation developers. I like to assume that the developer is a junior/mid-weight, and/or is new-ish to Umbraco.

In my experience lines like this can confuse new developers:
composition.AddEnvironmentDashboard(new Dictionary<DashboardEnvironment, IEnumerable<string>>

The 2 main things which I would like to be very obvious:
1. Registration of the pair: environment name + domain
2. Registration of additional panels/groups on the dashboard (eg. uptime, db server)

So, I would like to change DashboardEnvironment to DomainInfo which we previously had, as that seems more obvious IMO for registering domains.


I'm wondering if something like the following will be easier to understand for new devs. It's not as "elegant" for more advanced developers:

var myDashboard = new EnvironmentDashboard();
myDashboard.Domains.Add(new DomainInfo("Local", "localhost:21801"));
myDashboard.Domains.Add(new DomainInfo("Development", "environmentdashboard-dev.ry.com"));
myDashboard.Domains.Add(new DomainInfo("QA", "environmentdashboard-qa.ry.com"));
myDashboard.Domains.Add(new DomainInfo("UAT", "environmentdashboard-uat.ry.com"));
myDashboard.Domains.Add(new DomainInfo("Production", "environmentdashboard.ry.com"));
myDashboard.Groups.Add(new UptimeDashboardGroup())
myDashboard.Groups.Add(new DatabaseServerGroup());

composition.AddEnvironmentDashboard(myDashboard);

Reply:
It kan be made more and easier, but I only used a couple of hours putting it together as sample, so it can be refactored a lot to make it easier to understand.

I didn't use the DomainInfo as I would have to make the urls a collection as in my experience a single site can have multiple domains attached.

eg. an Azure site has the Azure url and the actual production url.

And I liked the thought of the Environment to be the dominant part, not the Domain as the DomainInfo implies.

Reply:
hmm... I think that a list of DomainInfo can handle that.

Regisiter: 
"Production" + azurewebsites domain
"Production" + pre-live domain
"Production" + real live domain

Reply:
it can but then its the domain that's the main thing not the Environment

If you ever want to attach dashboard groups per environment its more intuitive that the Environment is main entry point and that a single environment can have multiple domains.

Reply:
Ok single entry point makes sense. Environment has domains.

Probably just call it Environment then? Not DashboardEnvironment. Unless there is a better name?

Reply:
I used DashboardEnvironment as its kind of annoying to workaround System.Environment

Maybe Deployment or Workspace could be naming options, wasn't Workspace the name in used by Umbraco in the Browser Title?